obspy/core/stream.py

```diff
@@ -3118,6 +3119,80 @@ def remove_sensitivity(self, *args, **kwargs):
             tr.remove_sensitivity(*args, **kwargs)
         return self
 
+    def stack(self, group='all', type='linear', npts_tol=0):
```

Comment:
Could you rename `type` to `stack_type` to not clash with the reserved `type` word?

Reply:
Also maybe rename `group` to `group_by` which is what most other packages like pandas would probably call an operation like this.